contrib/babelfishpg_common/src/geo.c

```diff
@@ -0,0 +1,175 @@
+#include "postgres.h"
+#include "fmgr.h"
+#include <string.h>
+#include "utils/geo_decls.h"
+#include "utils/builtins.h"
+#include "geo_data.h"
+#include "lib/stringinfo.h"
+#include <stdlib.h>
+#include <stdio.h>
+
+#define FLOAT8_TO_CSTRING(x)         DatumGetCString(DirectFunctionCall1(float8out, Float8GetDatum(x)))
+#define YYFREE                       pfree
+
+#define POINTFLAG_Z                  (1 << 0)  
+#define POINTFLAG_M                  (1 << 1)
+
+#define FLAGS_SET_Z(flags, value)    ((value) ? ((flags) |= POINTFLAG_Z) : ((flags) &= ~POINTFLAG_Z))
+#define FLAGS_SET_M(flags, value)    ((value) ? ((flags) |= POINTFLAG_M) : ((flags) &= ~POINTFLAG_M))
+
+#define FLAGS_GET_Z(flags)           ((flags) & POINTFLAG_Z)
+#define FLAGS_GET_M(flags)           ((flags) & POINTFLAG_M)
+
+PG_FUNCTION_INFO_V1(geo_wkt_rewrite);
+
+Datum
+geo_wkt_rewrite(PG_FUNCTION_ARGS)
```

Comment:
Why is this a Datum function? All I see that this function is being used only in spatialtypes.c file, which means we can just create a normal function.